libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -597,9 +597,20 @@ void Compass::_detect_backends(void)
         ADD_BACKEND(DRIVER_LIS3MDL, AP_Compass_LIS3MDL::probe(*this, hal.i2c_mgr->get_device(1, HAL_COMPASS_LIS3MDL_I2C_ADDR),
                                                               true, ROTATION_YAW_90),
                     AP_Compass_LIS3MDL::name, true);
+
         ADD_BACKEND(DRIVER_LIS3MDL, AP_Compass_LIS3MDL::probe(*this, hal.i2c_mgr->get_device(0, HAL_COMPASS_LIS3MDL_I2C_ADDR),
                                                               both_i2c_external, both_i2c_external?ROTATION_YAW_90:ROTATION_NONE),
                     AP_Compass_LIS3MDL::name, both_i2c_external);
+
+        // external lis3mdl in mRo Neo-M8N GPS module
+        ADD_BACKEND(DRIVER_LIS3MDL, AP_Compass_LIS3MDL::probe(*this, hal.i2c_mgr->get_device(1, HAL_COMPASS_LIS3MDL_I2C_ADDR),
```

Comment:
Thanks, I got confused when trying to figure out what the rotation should be and forgot to merge those lines... Since the mRo sensor can be set to either I2C address there will be ambiguity between it and PH2 if those values need to be different. Guess I'll need to understand why it didn't work properly when I used "ROTATION_NONE"; maybe I wound up with 2 backends for the same sensor...